lambdas/es/indexer/index.py

```diff
@@ -187,18 +187,42 @@ def now_like_boto3():
     return datetime.datetime.now(tz=tzutc())
 
 
-def infer_extensions(key, ext):
+def get_compression(ext: str):
+    """return the compression type or None if not supported"""
+    return "gz" if ext == ".gz" else None
+
+
+def get_normalized_extensions(key) -> List[str]:
+    """standard function turning keys into a list of (possibly empty) extensions"""
+    path = pathlib.PurePosixPath(key)
+    try:
+        ext_last = path.suffix.lower()
+        ext_next_last = path.with_suffix('').suffix.lower()
+    except ValueError:
+        return ["", ""]
+
+    # return in left-to-right order as they occur in the key
+    return [ext_next_last, ext_last]
+
+
+def infer_extensions(key, exts: List[str], compression):
     """guess extensions if possible"""
     # Handle special case of hive partitions
     # see https://www.qubole.com/blog/direct-writes-to-increase-spark-performance/
+    long_ext = "".join(exts)
+    # pylint: disable=too-many-boolean-expressions)
     if (
-            re.fullmatch(r".c\d{3,5}", ext) or re.fullmatch(r".*-c\d{3,5}$", key)
+            re.fullmatch(r".c\d{3,5}", long_ext) or re.fullmatch(r".*-c\d{3,5}$", key)
             or key.endswith("_0")
-            or ext == ".pq"
+            or exts[-1] == ".pq"
+            or compression and exts[0] == ".pq"
```

Comment:
Parenthesis around "and"?

Reply:
Will do. Same effect though because of `and` precedence over `or`.